sources/pojo/src/main/java/com/kaltura/client/utils/MetadataUtils.java

```diff
@@ -93,6 +100,76 @@ public static String getValue(String xml, String xPath) throws ParserConfigurati
 
 		return (String) compiledExpression.evaluate(doc, XPathConstants.STRING);
 	}
+	
+	public static List<String> getFields(String xsd) throws ParserConfigurationException, SAXException, IOException, XPathExpressionException
+	{
+		DocumentBuilderFactory docBuilderFactory = DocumentBuilderFactory.newInstance();
+		docBuilderFactory.setIgnoringElementContentWhitespace(true);
+
+		DocumentBuilder docBuilder = docBuilderFactory.newDocumentBuilder();
+		Document doc = docBuilder.parse(new ByteArrayInputStream(xsd.getBytes()));
+
+		NodeList nodeList = doc.getElementsByTagName("xsd:element");
+		List<String> list = new ArrayList<String>();
+		for (int i = 0; i < nodeList.getLength(); i++) {
+			Element element = (Element)nodeList.item(i);
+			String fieldName = element.getAttribute("name");
+			if (fieldName.equals("metadata"))
+				continue;
+			
+			list.add(fieldName);
+		}
+		
+		return list;
+		
+	}
+	
+	public static LinkedHashMap<String, ArrayList<String>> getValues(String xml) throws XPathExpressionException, ParserConfigurationException, SAXException, IOException
```

Comment:
Please change the returned type to Map<String, List<String>>, the implementation of these interfaces shouldn't matter.